### ISSUE TITLE
Timezone issues

### ISSUE THREAD

**Issue by Dzordzu** [anoadragon453/matrix-reminder-bot#63]
[This commit](https://github.com/anoadragon453/matrix-reminder-bot/commit/5d5457f570bb46f5c6c97c979c8c312f80656985) causes erorrs when using `!remindme every...`. Without `every` it works fine

Here are some associated issues
- https://github.com/celery/celery/issues/983
- https://stackoverflow.com/questions/10306389/python-pass-tzinfo-to-naive-datetime-without-pytz

**Comment by Dzordzu** [comment]
[This line](https://github.com/anoadragon453/matrix-reminder-bot/commit/5d5457f570bb46f5c6c97c979c8c312f80656985#diff-7c25f6bd8a58b0d2781d138ea517c48c29fb214115d003c00f9b8e69f9a4eed4R82) causes

```
reminder-bot_1      | 2020-11-17 17:40:32,742 | matrix_reminder_bot.callbacks [ERROR] Unknown error while processing command:
reminder-bot_1      | Traceback (most recent call last):
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/callbacks.py", line 71, in message
reminder-bot_1      |     await command.process()
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/bot_commands.py", line 291, in process
reminder-bot_1      |     await self._remind_me()
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/functions.py", line 81, in inner
reminder-bot_1      |     await command_func(self, *args, **kwargs)
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/bot_commands.py", line 319, in _remind_me
reminder-bot_1      |     await self._remind(target=self.event.sender)
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/bot_commands.py", line 268, in _remind
reminder-bot_1      |     reminder = Reminder(
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/matrix_reminder_bot/reminder.py", line 82, in __init__
reminder-bot_1      |     start_time = tz.localize(start_time)
reminder-bot_1      |   File "/usr/local/lib/python3.8/site-packages/pytz/tzinfo.py", line 318, in localize
reminder-bot_1      |     raise ValueError('Not naive datetime (tzinfo is already set)')
reminder-bot_1      | ValueError: Not naive datetime (tzinfo is already set)

```

**Comment by anoadragon453** [comment]
...and this is why test suites are useful kids.

**Comment by Dzordzu** [comment]
Are you working on it? Or should I make a PR?

**Comment by anoadragon453** [comment]
Working on it, though if you have a solution feel free to post it. I'm currently trying to figure out why this only occurs when using a recurring reminder.

**Comment by anoadragon453** [comment]
So this happens upon creating a reminder as we're handing `Reminder` a `start_time` with timezone information associated with it. Whereas on startup, the reminders we load from the database have no timezone information associated with them, and thus this exception does not occur.

The fix is to remind timezone information from `start_time` before constructing a new `Reminder`.

**Comment by Dzordzu** [comment]
Maybe instead of `start_time = tz.localize(start_time)` use `  start_time.astimezone(tz)`?

**Comment by anoadragon453** [comment]
That may work as well, but fundamentally `Reminder` shouldn't be given `start_time`s that have timezone information attached, so first I'm preventing that from happening.

**Comment by Dzordzu** [comment]
Do you have any CI/CD (or automated script) connected to the repo?

**Comment by anoadragon453** [comment]
> Do you have any CI/CD (or automated script) connected to the repo?

Yes, see https://github.com/anoadragon453/matrix-reminder-bot/actions.

**Comment by Dzordzu** [comment]
Super <3
I've managed to fix issue by using my suggestion

**Comment by Dzordzu** [comment]
![image](https://user-images.githubusercontent.com/17514622/99435043-df262780-290f-11eb-92b3-7060d87aed96.png)

**Comment by anoadragon453** [comment]
![image](https://user-images.githubusercontent.com/1342360/99435165-ada95e00-2907-11eb-990c-1f227b9dcc1b.png)

Same! Through fixing the issue at the other end (ensuring tz-aware datetimes are not passed to `Reminder`). Feel free to put up a PR with your fix and I'll merge it in :)

**Comment by anoadragon453** [comment]
@Dzordzu While #64 does fix this issue, I'm not opposed to merging #65 if you think it's still a useful change.

**Comment by Dzordzu** [comment]
"If it works, don't touch it". I will test it later, maybe do a little code refactoring and prepare a new, tested PR
